department-of-veterans-affairs/notification-api
# adding migration script to update templates with a reply_to field

# PR for story #605 

> Adding in a `migrations/versions/0342_reply_to_field.py` file to update the templates table with a `reply_to` field.
> This will allow us to store specific reply to emails per template if needed for the user. 

# Test instructions

> Pull branch local.
> Login to the DB `psql notification_api postgres`.
> Check the templates DB table for column names: 
>  ``` 
> select column_name, data_type, is_nullable from information_schema.columns where table_name = 'templates';
> ```

```
notification_api=# select column_name, data_type, is_nullable from information_schema.columns where table_name = 'templates';
        column_name        |          data_type          | is_nullable 
---------------------------+-----------------------------+-------------
 communication_item_id     | uuid                        | YES
 archived                  | boolean                     | NO
 service_letter_contact_id | uuid                        | YES
 hidden                    | boolean                     | NO
 provider_id               | uuid                        | YES
 id                        | uuid                        | NO
 template_type             | USER-DEFINED                | NO
 created_at                | timestamp without time zone | NO
 updated_at                | timestamp without time zone | YES
 service_id                | uuid                        | NO
 created_by_id             | uuid                        | NO
 version                   | integer                     | NO
 name                      | character varying           | NO
 process_type              | character varying           | NO
 subject                   | text                        | YES
 postage                   | character varying           | YES
 content                   | text                        | NO
(17 rows)
```

> See that the column `reply_to_email` is not in the table.
> Run `flask db upgrade` to update the local Postgres DB with the latest migration script.
> Check the templates DB table for column names:
>  ``` 
> select column_name, data_type, is_nullable from information_schema.columns where table_name = 'templates';
> ```
> Validate that the `reply_to_email` field exists.
```
notification_api=# select column_name, data_type, is_nullable from information_schema.columns where table_name = 'templates';
        column_name        |          data_type          | is_nullable 
---------------------------+-----------------------------+-------------
 id                        | uuid                        | NO
 version                   | integer                     | NO
 archived                  | boolean                     | NO
 service_letter_contact_id | uuid                        | YES
 hidden                    | boolean                     | NO
 provider_id               | uuid                        | YES
 communication_item_id     | uuid                        | YES
 template_type             | USER-DEFINED                | NO
 created_at                | timestamp without time zone | NO
 updated_at                | timestamp without time zone | YES
 service_id                | uuid                        | NO
 created_by_id             | uuid                        | NO
 name                      | character varying           | NO
 subject                   | text                        | YES
 process_type              | character varying           | NO
 postage                   | character varying           | YES
 content                   | text                        | NO
 reply_to_email            | text                        | YES
(18 rows)
```

# Release Instructions

None.


# Reviewer checklist 

This is a suggested checklist of questions reviewers might ask during their
review
